pkg/prometheus/promcfg.go

```diff
@@ -4026,6 +4026,56 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 		})
 	}
 
+	// OVHCloudSDConfigs
+	if len(sc.Spec.OVHCloudSDConfigs) > 0 {
+		configs := make([][]yaml.MapItem, len(sc.Spec.OVHCloudSDConfigs))
+		for i, config := range sc.Spec.OVHCloudSDConfigs {
+			configs[i] = append(configs[i], yaml.MapItem{
+				Key:   "application_key",
+				Value: config.ApplicationKey,
+			})
+
+			value, _ := store.GetSecretKey(ctx, sc.GetNamespace(), *config.ApplicationSecret)
+			configs[i] = append(configs[i], yaml.MapItem{
+				Key:   "application_secret",
+				Value: value,
+			})
+
+			key, _ := store.GetSecretKey(ctx, sc.GetNamespace(), *config.ConsumerKey)
+			configs[i] = append(configs[i], yaml.MapItem{
+				Key:   "consumer_key",
+				Value: key,
+			})
+
+			if config.Service != nil {
+				configs[i] = append(configs[i], yaml.MapItem{
+					Key:   "service",
+					Value: config.Service,
```

Comment:
Thanks. 
Is it more reasonable to maintain the user-defined capitalization specifications of CRD? 🤔️

Reply:
> Is it more reasonable to maintain the user-defined capitalization specifications of CRD? 🤔️

Not sure to follow here, could you elaborate?

Reply:
> Not sure to follow here, could you elaborate?

ref: 
https://github.com/prometheus/prometheus/blob/5c417684f87f2e7f55b172b3e5e5c96ac9bc2e2c/discovery/ovhcloud/ovhcloud.go#L89-L94

For me, remove the enum limit. It's best to keep it vague. 
/cc @simonpasquier @slashpai

Reply:
> It's best to keep it vague.

Not sure what you mean but my opinion is, if we can add valuable validation at the operator its best we do that. I'd go a step further and say its best if we can add most of the validations at the API level (i.e, make the API restrictive) and only touch the application level validation when absolutely necessary. e.g, when fetching secrets, etc.

TLDR; If Prometheus is validating for something (at the application level or otherwise) it is best that the operator do at least that. I think @simonpasquier and @slashpai will agree as well.